### ISSUE TITLE
Emit @keyup, @keydown, @click, @focus, @blur on FormulateInput from native inputs

### ISSUE THREAD

**Issue by soullpoint** [wearebraid/vue-formulate#86]
**Describe the new feature you'd like**
I would like to be able to listen for keypress events on a FormulateInput component. Perhaps I'm doing something wrong, but 

```
<FormulateInput
        type="text"
        placeholder="Test"
        v-model="test"
       @keyup.tab="someFunction"
/>
```

does not trigger `someFunction` when the tab key is pressed. This is not limited to this key, I have tried other keys as well.

**What percentage of vue-formulate users would benefit?**
I don't know if this is a common use case, but in my case, I want to be able to trigger the creation of a new item when tab is pressed.

**Comment by andrew-boyd** [comment]
@soullpoint so a couple of things here. For one, `tab` has native behavior in forms (to advance to the next item in the tab index) so it may be inadvisable to bind to that key for accessibility reasons. 

TLDR; https://codepen.io/boyd/pen/ZEbqwBY

That said, it's possible for sure, but for a `@keyup` to fire on the element you're targeting you'd have to tab into it (that's the `@keydown`) and release tab while it's focused (that's the `@keyup`) otherwise tab is going to advance you to the next item in the tab index before your `@keyup` fires. Using `@keydown` instead will give you the behavior you're looking for — the user is focused on a field and presses the tab key which then fires your own custom method. You can also add the `.prevent` modifier to keep the native tab behavior from firing.

The last thing to consider is that when you're adding event modifiers such as `@keydown` or `@keyup` to a component and not a native HTML element you'll want to add the `.native` modifier to tell Vue that you're looking for the native DOM event not a custom event being emitted by the component to any listening parent.

`@keydown.tab.native.prevent="someFunction"`

I've put together an example codepen that shows the behavior on a `FormulateInput` contrasted with a native HTML `input` element.

https://codepen.io/boyd/pen/ZEbqwBY

**Comment by justin-schroeder** [comment]
Good summary @andrew-boyd. I will say we should consider how to make this pass through more easy. We've done this on a button's in 2.3 by just re-emitting the original event so people can use a standard `@click` event on buttons.

**Comment by andrew-boyd** [comment]
@justin-schroeder Given that it's possible to bind the events using the `.native` modifier I'm unclear on what more should / could be done. Should we re-open this issue and discuss it?

**Comment by justin-schroeder** [comment]
Yeah, the fact that all this works with the `.native` modifier makes it a much lower priority, but we've had several of these issues opened where there was a developer expectation of native event emission, which I think is a fair assumption. The idea would be to collect all the events on the native input and re-emit them from the `<FormualteInput />` itself. There are times/reasons where this might be really desirable, because the `.native` modifier is technically still listening for events on the wrapping div, not the input itself.

**Comment by andrew-boyd** [comment]
Ok, Re-opening. Good point about these events being on the wrapper and not the actual deeply nested input elements themselves.

**Comment by justin-schroeder** [comment]
Either way, the codepen is a great documentation piece for people who struggle with this in the meantime.

**Comment by soullpoint** [comment]
@andrew-boyd Thank you for the example, and I really appreciate the advice in regards to accessibility. I'm rather new to this so things like this really go a long way.

Keep up the good work!

**Comment by soullpoint** [comment]
@andrew-boyd This might somewhat unrelated, I am trying to focus a form element when it is rendered (by the "tab" keypress event I was struggling with before).

Currently I have it working as follows (excuse the TypeScript casting):
```
...

mounted() {
    (this.$refs.text as Vue).$el.querySelector("input")?.focus();
}

...
```

This works as expected, however, I was wondering if this is the best way to do it? I tried setting the autofocus attribute on the FormulateInput component without any luck.

I appreciate that you're likely busy so thank you for your time.

**Comment by andrew-boyd** [comment]
@soullpoint I think that's a perfectly valid way to set focus when your component mounts.

**Comment by silvio-e** [comment]
It would be also very awesome to have the `@focus` event available. I use it usually for checks for making floating labels work. Or do you have any other recommended way of achieving this? Actually it would be awesome if floating labels could be an included option in your theme :)

**Comment by justin-schroeder** [comment]
@silvio-e Currently we do this on our own projects where we use floating labels we prefer "floating" the label when the user starts entering content rather than focus ([we use the  `data-has-value` property](https://vueformulate.com/guide/theming/styling-tips/#data-attributes) on the outer wrapper). However — we agree! We should definitely forward the `focus` and `blur` events events up.

**Comment by silvio-e** [comment]
That's great! :) These floating labels are a requirement for forms I would like to use formulate for. So I guess I wait for a bit. Could you indicate by when this could available?

I have just tested the schema and I love it!! :) Came just to my mind that we need the events implemented there too O_o :D

But at least for my floating label case a `data-has-focus` would also work :) what do you think about that?

**Comment by silvio-e** [comment]
Ah and btw. `@blur` seems to work already!

**Comment by justin-schroeder** [comment]
I think `data-has-focus` is probably a good idea. That makes it something we need to add to the all-powerful [context object](https://vueformulate.com/guide/inputs/#context-object) 👀 but thats ok. I think you guys make a good case that it's part of the context 👍

**Comment by justin-schroeder** [comment]
This is now published in `v2.4.3`

**Comment by silvio-e** [comment]
Awesome!!! 🎉  Thanks a lot! 😃 

Do you think you are going to implement `data-has-focus` also? That would come handy when working with the schema 😃

**Comment by justin-schroeder** [comment]
@silvio-e I was no longer planning to since events are supported, but I will consider it! Feel free to open a new feature request issue. Thanks!

**Comment by thunderwin** [comment]
@blur give me a   `{"isTrusted":true} `. ..  why..
